### PR TITLE
Refactor SelectOption to use Hooks

### PR DESCRIPTION
**Pull request by patelrr**
#### What does this PR do?
Refactor `src/js/components/Select/SelectOption`

#### Where should the reviewer start?
`src/js/components/Select/SelectOption`

#### What testing has been done on this PR?
Running `yarn storybook` to check if it behaves as expected

#### How should this be manually tested?
Run `yarn storybook` and check its behaviour

#### What are the relevant issues?
[#3394 ](https://github.com/grommet/grommet/issues/3394)

#### Screenshots (if appropriate)
N/A

#### Do the grommet docs need to be updated?
No

#### Is this change backwards compatible or is it a breaking change?
No

### CODE REVIEW

**Comment by ShimiSun** [comment]
Hi buddy! thanks for the PR, do you mind taking a look at what caused the breaking tests? they should be intact post refactoring.

**Comment by patelrr** [comment]
Hi @ShimiSun, I got same test failed on my local system. But after running `yarn test -u` (as suggested in test terminal) it got fixed. Can you suggest how to fix it?

**Comment by ShimiSun** [comment]
Yes, push the updated snapshots to be part of your PR

**Comment by patelrr** [comment]
Pushed

File: src/js/components/Select/SelectOption.js [hunk 1]
```diff
@@ -1,42 +1,39 @@
...
-  }
+function shouldComponentUpdate(prevProps, nextProps) {
+  const { active, disabled, option, selected } = prevProps;
+  const {
+    active: nextActive,
+    disabled: nextDisabled,
+    option: nextOption,
+    selected: nextSelected,
+  } = nextProps;
+  return (
```
Comment (husseyexplores) [inline src/js/components/Select/SelectOption.js]: `React.memo` actually takes `areEqual` as second argument. It is the opposite of `shouldComponentUpdate`.
Reference: https://reactjs.org/docs/react-api.html#reactmemo
```suggestion
return (
  active === nextActive &&
  selected === nextSelected && 
  disabled === nextDisabled && 
  option === nextOption 
  );
```
You can also update the function name to match the context here.

Comment (patelrr) [reply src/js/components/Select/SelectOption.js]: Thanks. Fixed it.

**Pull request merged**